##Customer segmentation data from Kaggle

An automobile company has plans to enter new markets with their existing products (P1, P2, P3, P4 and P5). After intensive market research, they’ve deduced that the behavior of new market is similar to their existing market.

In their existing market, the sales team has classified all customers into 4 segments (A, B, C, D ). Then, they performed segmented outreach and communication for different segment of customers. This strategy has work exceptionally well for them. They plan to use the same strategy on new markets and have identified 2627 new potential customers.

We are required to help the manager to predict the right group of the new customers.

In [1]:
import pandas as pd

In [2]:
df = pd.read_csv('/content/drive/MyDrive/codingDojo/data_science/project_2/cars_Train.csv')
df.head()

ID  Gender Ever_Married  ...  Family_Size  Var_1 Segmentation
0  462809    Male           No  ...          4.0  Cat_4            D
1  462643  Female          Yes  ...          3.0  Cat_4            A
2  466315  Female          Yes  ...          1.0  Cat_6            B
3  461735    Male          Yes  ...          2.0  Cat_6            B
4  462669  Female          Yes  ...          6.0  Cat_6            A

[5 rows x 11 columns]

This is a classification problem with target "Segmentation" column (A, B, C, D ).

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8068 entries, 0 to 8067
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   ID               8068 non-null   int64  
 1   Gender           8068 non-null   object 
 2   Ever_Married     7928 non-null   object 
 3   Age              8068 non-null   int64  
 4   Graduated        7990 non-null   object 
 5   Profession       7944 non-null   object 
 6   Work_Experience  7239 non-null   float64
 7   Spending_Score   8068 non-null   object 
 8   Family_Size      7733 non-null   float64
 9   Var_1            7992 non-null   object 
 10  Segmentation     8068 non-null   object 
dtypes: float64(2), int64(2), object(7)
memory usage: 693.5+ KB


This datasrt has 11 features and 8068 rows of data.

In [4]:
df.isnull().sum()

ID                   0
Gender               0
Ever_Married       140
Age                  0
Graduated           78
Profession         124
Work_Experience    829
Spending_Score       0
Family_Size        335
Var_1               76
Segmentation         0
dtype: int64

In [6]:
df.duplicated().sum()

0

In [7]:
df['Segmentation'].value_counts()

D    2268
A    1972
C    1970
B    1858
Name: Segmentation, dtype: int64

- The dataset is pretty balanced.
- Most features are categorical.
- Missing data.